### PR TITLE
Extract task_receiver from stf_task_thread

### PR DESCRIPTION
**Pull request by ziming-zung**
This PR:
1.  Make stf_task(stf_task_sender/receiver/handler) naming clearer
2.  Extract task_receiver

If you noticed the `TrustedCall::litentry_trusted_call` code, please ignore it

### CODE REVIEW

File: litentry/core/stf-task/receiver/src/stf_task_receiver.rs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+			.recv()
+			.map_err(|e| Error::OtherError(format!("receiver error:{:?}", e)))?;
+
+		match request_type {
+			RequestType::Web2IdentityVerification(ref request) => {
+				if let Err(e) = web2_identity_verification(&request_context, request.clone()) {
+					error!("Could not retrieve data from https server due to: {:?}", e);
+				}
+			},
+			RequestType::Web3IndentityVerification(ref _request) => {
```
Comment (Kailai-Wang) [inline litentry/core/stf-task/receiver/src/stf_task_receiver.rs]: Just notice the typo here (and the `RequestType` definition)

File: app-libs/stf/src/stf_sgx_litentry.rs [hunk 2]
```diff
@@ -171,14 +184,14 @@ impl Stf {
...
-		let request = lc_stf_task_handler::Web2IdentityVerificationRequest {
+		let request = Web2IdentityVerificationRequest {
 			target,
 			identity,
 			challenge_code: code.unwrap(),
 			validation_data,
 			bn,
 		};
-		let xt_sender = lc_stf_task_handler::stf_task_sender::StfRequestSender::new();
+		let xt_sender = StfRequestSender::new();
```
Comment (Kailai-Wang) [inline app-libs/stf/src/stf_sgx_litentry.rs]: maybe we can simply rename it to `sender`

**Review by Kailai-Wang — APPROVED** [review]
Looks good except for a few small things, we need to make clippy happy too

File: litentry/core/stf-task/receiver/src/stf_task_receiver.rs (hunk 1 above)
Comment (ziming-zung) [reply litentry/core/stf-task/receiver/src/stf_task_receiver.rs]: done.

File: app-libs/stf/src/stf_sgx_litentry.rs (hunk 2 above)
Comment (ziming-zung) [reply app-libs/stf/src/stf_sgx_litentry.rs]: done

**Pull request merged**